blame/blame_test.go

```diff
@@ -0,0 +1,810 @@
+package blame
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/src-d/go-git.v2"
+	"gopkg.in/src-d/go-git.v2/core"
+	"gopkg.in/src-d/go-git.v2/formats/packfile"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type SuiteCommon struct {
+	repos map[string]*git.Repository
+}
+
+var _ = Suite(&SuiteCommon{})
+
+var fixtureRepos = [...]struct {
+	url      string
+	packfile string
+}{
+	{"https://github.com/tyba/git-fixture.git", "../formats/packfile/fixtures/git-fixture.ofs-delta"},
+	{"https://github.com/spinnaker/spinnaker.git", "../formats/packfile/fixtures/spinnaker-spinnaker.pack"},
+}
+
+// create the repositories of the fixtures
+func (s *SuiteCommon) SetUpSuite(c *C) {
+	s.repos = make(map[string]*git.Repository, 0)
+	for _, fixRepo := range fixtureRepos {
+		s.repos[fixRepo.url] = git.NewPlainRepository()
+
+		d, err := os.Open(fixRepo.packfile)
+		defer func() {
+			c.Assert(d.Close(), IsNil)
+		}()
+		c.Assert(err, IsNil)
+
+		r := packfile.NewReader(d)
+		r.Format = packfile.OFSDeltaFormat // TODO: how to know the format of a pack file ahead of time?
```

Comment:
The pktline we get before the packfile specifies which capabilities the packfile does support.